airflow/plugins/operators/littlepay_raw_sync.py

```diff
@@ -0,0 +1,217 @@
+import concurrent
+import json
+import os
+import traceback
+from concurrent.futures import Future, ThreadPoolExecutor
+from datetime import datetime
+from typing import ClassVar, Dict, List
+
+import boto3
+import pendulum
+from calitp_data_infra.auth import get_secret_by_name
+from calitp_data_infra.storage import (
+    PARTITIONED_ARTIFACT_METADATA_KEY,
+    PartitionedGCSArtifact,
+    get_fs,
+    get_latest_file,
+)
+from pydantic.class_validators import validator
+from pydantic.error_wrappers import ValidationError
+from pydantic.main import BaseModel
+from tqdm import tqdm
+from tqdm.contrib.logging import logging_redirect_tqdm
+
+from airflow.models import BaseOperator
+
+LITTLEPAY_RAW_BUCKET = os.getenv("CALITP_BUCKET__LITTLEPAY_RAW")
+
+
+class LittlepayFileKey(str):
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v):
```

Comment:
Could this maybe have a more specific name, like validate_filename, to avoid the implication that something about the file _contents_ is being validated if/when this is called from other places?